app/controllers/api/hackatime/v1/hackatime_controller.rb

```diff
@@ -240,6 +240,9 @@ def handle_heartbeat(heartbeat_array)
 
       parsed_ua = WakatimeService.parse_user_agent(heartbeat[:user_agent])
 
+      # Avoid unviewable categories
+      heartbeat[:category] ||= "unknown"
```

Comment:
This should be unset in the database & then in places we filter and show include a "unknown" choice